enigma-swing/src/main/java/org/quiltmc/enigma/gui/Gui.java

```diff
@@ -460,10 +461,13 @@ public void toggleMapping(EditorPanel editor) {
 	}
 
 	public void toggleMappingFromEntry(Entry<?> obfEntry) {
-		if (this.controller.getProject().getMapper().getDeobfMapping(obfEntry).targetName() != null) {
+		EntryMapping mapping = this.controller.getProject().getMapper().getMapping(obfEntry);
+
+		// todo figure out how to toggle back onto proposed - this will toggle back to obf
```

Comment:
unsure if this is actually worth the bother -- we'd have to store proposed names separately to do this, and it's not really intuitive behaviour unless you're already used to it

Reply:
How do you get a proposed name back then?

Reply:
I haven't really thought of a good solution